libs/jsonschema/from_type.go

```diff
@@ -0,0 +1,338 @@
+package jsonschema
+
+import (
+	"container/list"
+	"fmt"
+	"maps"
+	"path"
+	"reflect"
+	"slices"
+	"strings"
+)
+
+// Fields tagged "readonly" should not be emitted in the schema as they are
+// computed at runtime, and should not be assigned a value by the bundle author.
+const readonlyTag = "readonly"
+
+// Annotation for internal bundle fields that should not be exposed to customers.
+// Fields can be tagged as "internal" to remove them from the generated schema.
+const internalTag = "internal"
+
+// Annotation for bundle fields that have been deprecated.
+// Fields tagged as "deprecated" are omitted from the generated schema.
+const deprecatedTag = "deprecated"
```

Comment:
All these indicate a field should be skipped. Why separate variables instead of a slice?

Reply:
done